src/test/java/com/google/googleidentity/filter/UserAuthenticationFilterTest.java

```diff
@@ -61,26 +51,7 @@ public void testFilter_noUserPresent_redirectAndSetOldUrl()
 
         HttpServletRequest request = mock(HttpServletRequest.class);
         HttpServletResponse response = mock(HttpServletResponse.class);
-        HttpSession httpSession = mock(HttpSession.class);
-        Map<String, Object> sessionMap = new HashMap<>();
-
-        Mockito.doAnswer(new Answer() {
-            @Override
-            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
-                String key = (String) invocationOnMock.getArguments()[0];
-                return sessionMap.get(key);
-            }
-        }).when(httpSession).getAttribute(anyString());
-
-        Mockito.doAnswer(new Answer() {
-            @Override
-            public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
-                String key = (String) invocationOnMock.getArguments()[0];
-                Object value = invocationOnMock.getArguments()[1];
-                sessionMap.put(key, value);
-                return null;
-            }
-        }).when(httpSession).setAttribute(anyString(), anyObject());
+        HttpSession httpSession = new FakeHttpSession();
```

Comment:
nit: if this local variable is typed "FakeHttpSession" instead of "HttpSession" then you can avoid the cast below. It's fine to keep the stronger typing in the test so that test-specific methods on FakeHttpSession are easier to use.